core/src/main/java/org/infinispan/commands/CommandUUID.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.commands;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.marshall.core.Ids;
+import org.infinispan.remoting.transport.Address;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Represents an unique identified for non-transaction write commands.
+ *
+ * It is used to lock the key for a specific command.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public class CommandUUID {
+
+   private static final AtomicLong nextId = new AtomicLong(0);
+
+   private final Address address;
+   private final long id;
+
+   private CommandUUID(Address address, long id) {
+      this.address = address;
+      this.id = id;
+   }
+
+   public static CommandUUID generateUUID(Address address) {
```

Comment:
Is it worth it? I mean, create a global component will take memory that does not need to be wasted.